pkg/asset/agent/manifests/infraenv.go

```diff
@@ -70,6 +71,11 @@ func (i *InfraEnv) Generate(dependencies asset.Parents) error {
 				},
 			},
 		}
+
+		// Use RpmArch for infraEnv.Spec.CpuArchiteture. Default is amd64 or "", needs to be x86_64 or aarch64
+		if installConfig.Config.ControlPlane.Architecture != "" {
+			infraEnv.Spec.CpuArchitecture = arch.RpmArch(string(installConfig.Config.ControlPlane.Architecture))
```

Comment:
Elsewhere it looks like casting the argument to `RpmArch` to a string hasn't been required. Maybe we can drop it here too?
I know this is a nitpick but I am still a little traumatised from #6855.